File: solutions/dart/leap/1/lib/leap.dart
class Leap {
  bool leapYear(int year) {
    var result = false;
    if(year % 4 == 0) {
      if(year % 100 == 0) {
        if(year % 400 == 0) {
          result = true;
        }
      } else {
        result = true;
      }
    }
    return result;
  }
}

File: solutions/cpp/leap/1/leap.cpp
#include "leap.h"

namespace leap {
    bool is_leap_year(const unsigned int year) {
        bool result = false;
        if(year % 400 == 0 || (year % 4 == 0 && year % 100 != 0)) {
            result = true;
        }
        return result;
    }
}  // namespace leap
